src/html_support_files/odoc.css

```diff
@@ -550,12 +550,21 @@ div.odoc-spec,.odoc-include {
   min-width: 40%;
 }
 
+.spec.type .poly-variant > code {
+  min-width: 2ch;
+}
+.spec.type .poly-variant > code + code {
+  min-width: calc(40% - 2ch - 0.6ex);
+  /* In polymorphic variants, the | is not included in the first code block, so we
+   need to accound for its size */
+}
+
```

Comment:
Yes, this is fragile (and was already).

The reason the html generated in case of polymorphic variant is structured differently from the one for variants, is that it is already the case at the intermediate document representation.

I was afraid of changing the intermediate representation, as that would require me to check that it does not break anything in other backends. But maybe that is a better solution.